libs/platform/ebpf_serialize.h

```diff
@@ -0,0 +1,92 @@
+/*
+ *  Copyright (c) Microsoft Corporation
+ *  SPDX-License-Identifier: MIT
+ */
+
+// Data structures and functions for serializing and de-serializing eBPF sructs
+// between API and Execution Context.
+
+#pragma once
+
+#include "ebpf_platform.h"
+#include "ebpf_core_structs.h"
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+    /**
+     * @brief eBPF Core Map Information
```

Comment:
what does "core" mean here?

Reply:
I wanted a different type from the public ebpf_map_information. The key difference is ebpf_map_information uses char* to store pin_path whereas the core_ flavor stores it as counted string (ebpf_utf8_string_t).The "core" is a reference to the core libraries. I could rename it to _ebpf_platform_map_information or _ebpf_map_information_internal. Let me know if you have a preference.

Reply:
My first preference would be to not use a different type, which introduces the chance of bugs.  For example counted strings allow for there to be embedded nulls in the middle of a string, which is usually not a good thing.

Reply:
The entire pinning object module is based on ebpf_utf8_string_t. I could file an issue to address this separately.

Reply:
Filing a separate issue sounds fine.  I'd probably use _internal in the meantime then, which seems more accurate based on discussion above